### PR TITLE
Fix iOS build with Unity 2019.3 and later

### LINKED ISSUE #450: Undefined symbol: swiftCompatibility
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-sdk-for-unity/issues)
- [x] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [x] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Unity Editor Version: `2018.4.23`
- Unity SDK Version: `7.21.0`
- Installation Platform & Verison: iOS

## Goals

I want to be able to update Facebook SDK without constant problem solving.

## Expected Results

XCode would cleanly build the client after updating the SDK.

## Actual Results

XCode fails to build

## Steps to Reproduce

Update from 7.20.0 -> 7.21.0 and build on XCode

## Code Samples & Details

Showing Recent Messages
Undefined symbol: __swift_FORCE_LOAD_$_swiftCompatibilityDynamicReplacements
Undefined symbol: __swift_FORCE_LOAD_$_swiftCompatibility50

### PR DESCRIPTION
**Pull request by Nezz**
Thanks for proposing a pull request!

To help us review the request, please complete the following:

- [x] sign [contributor license agreement](https://developers.facebook.com/opensource/cla)
- [x] I've ensured that all existing tests pass and added tests (when/where necessary)
- [x] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [ ] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

I don't have the permission to add labels.

## Pull Request Details

The proposed fix only worked for older Unity versions. Resolves #450.

## Test Plan

Build for iOS using various Unity versions

### CODE REVIEW

File: Facebook.Unity.Editor/iOS/FixupFiles.cs [hunk 1]
```diff
@@ -53,7 +53,12 @@ public static void AddBuildFlag(string path)
             string projPath = Path.Combine(path, Path.Combine("Unity-iPhone.xcodeproj", "project.pbxproj"));
             PBXProject proj = new PBXProject();
             proj.ReadFromString(File.ReadAllText(projPath));
-            string targetGUID = proj.TargetGuidByName("Unity-iPhone");
+
+            // When building with Unity 2019.3 and later versions, there are two Xcode targets generated.
+            // The UnityFramework target will include the code and Unity-iPhone is only a thin wrapper around it.
+            // With earlier versions everything is in a single Unity-iPhone target.
+            var targetName = GetUnityVersionNumber() >= 201930 ? "UnityFramework" : "Unity-iPhone";
+            string targetGUID = proj.TargetGuidByName(targetName);
```
Comment (Thaina) [inline Facebook.Unity.Editor/iOS/FixupFiles.cs:61]: Will this better?

```C#
#if UNITY_2019_3_OR_NEWER
		string targetGUID = proj.GetUnityMainTargetGuid();
#else
		string targetName = PBXProject.GetUnityTargetName();
		string targetGUID = proj.TargetGuidByName(targetName);
#endif
```

**Review by Thaina — CHANGES_REQUESTED** [review]

File: Facebook.Unity.Editor/iOS/FixupFiles.cs (hunk 1 above)
Comment (Nezz) [reply Facebook.Unity.Editor/iOS/FixupFiles.cs:61]: No, that won't work. We cannot use preprocessor directives here as this file is compiled into a DLL that has to support various Unity versions.

Comment (Thaina) [reply Facebook.Unity.Editor/iOS/FixupFiles.cs:61]: @Nezz I see but still didn't `GetUnityFrameworkTargetGuid` and `GetUnityTargetName` better than constant?

Comment (Nezz) [reply Facebook.Unity.Editor/iOS/FixupFiles.cs:61]: `GetUnityFrameworkTargetGuid` is only available in Unity 2019.3 and later. If you include this DLL in a 2018.4 project, you will get a compiler error.

`GetUnityTargetName` was removed in Unity 2019.3. If you include this DLL in a 2019.3 project, you will get a compiler error.

Comment (Thaina) [reply Facebook.Unity.Editor/iOS/FixupFiles.cs:61]: Thanks you for clarification

**Pull request closed without merging**